Polaris/src/main.cpp

```diff
@@ -20,91 +24,149 @@
-
-Metro timer = Metro(1000/ LOOP_RATE);
-
-State * state;
 
 Sensorboard sensorBoard;
-
-void setup() {
-	Serial.begin(115200);
-
-	while(!Serial);
-	Wire.begin();
-	Wire.setClock(400000);
-
-	Serial.println("[Polaris] Initializing Sensor Board");
-	if(sensorBoard.setup()) {
-		Serial.println("[Polaris] Sensor Setup Complete!");
-	} else {
-		Serial.println("[Polaris] Sensor Setup Failed!");
-		while(1) {};
-	}
-
-
-	SPI.begin();
-    // SPI.beginTransaction(SPISettings(19000000, MSBFIRST, SPI_MODE0));
-	xbee->begin();
-	
-	servos = {
-		.paraServo_1 = new ServoController(PARACHUTE_SERVO_1), //double check direction 
-		.paraServo_2 = new ServoController(PARACHUTE_SERVO_2),
-		.paraServo_3 = new ServoController(PARACHUTE_SERVO_3),		
-		.paraServo_4 = new ServoController(PARACHUTE_SERVO_4),	
-		.cameraServo = new ServoController(CAMERA_SERVO),
-	}; 
-
-	flash->init();
-	int startAddress = 0;
-	startAddress = flash->rememberAddress();
-	Serial.println("Starting Flash Chip At Address: " + String(startAddress));
-
-	pinMode(IR_PIN, INPUT);
-	pinMode(PARACHUTE_SERVO_1_IN, INPUT); 
-	pinMode(PARACHUTE_SERVO_2_IN, INPUT); 
-	pinMode(PARACHUTE_SERVO_3_IN, INPUT); 
-	pinMode(PARACHUTE_SERVO_4_IN, INPUT); 
-
-
-	state = new PreLaunch(flash, attitudeStateEstimator, xbee, &servos, openMV);
-	state->initialize();
-
-	currentTime = millis();
-	previousTime = millis();
+AttitudeStateEstimator *attitudeStateEstimator = new AttitudeStateEstimator();
+
+//Payload Specific Stuff :)
+struct Servos servos;
+OpenMV *openMV = new OpenMV();
+
+State *state;
+
+
+#ifndef NO_SDCARD
+bool sdCardInitialized = false;
+File dataFile;
+#endif
+
+#ifndef NO_XBEE
+XbeeProSX *xbee = new XbeeProSX(30); // CS GPIO 17, use 30?!!!! - Kai 10/29/2024
+#endif
+
+void setup()
+{
+    Serial.begin(9600);
+
+#ifdef WAIT_FOR_SERIAL
+    while (!Serial)
+    {
+        yield()
+    }
+#endif
+
+    Wire.begin();
+    Wire.setClock(400000);
+
+    SPI.begin();
+
+#ifndef NO_SDCARD
+    // XXX: If using 5v only SD breakout board, make sure the first line is uncommented and the second line commented
+    if (SD.sdfs.begin(31, SPI_SIXTEENTH_SPEED))
+    { // REQUIRED for the non 3.3v tolerant SD breakout boards to work
+        int fileIdx = 0;
+        while (1)
+        {
+            char filename[100];
+            sprintf(filename, "flightData%d.bin", fileIdx++);
+            Serial.printf("Trying file `%s`\n", filename);
+            if (!SD.exists(filename))
+            {
+                dataFile = SD.open(filename, O_WRONLY | O_CREAT);
+                break;
+            }
+        }
+        sdCardInitialized = true;
+    }
+    else
+    {
+        Serial.println("SD Init failed");
+    }
+#endif
+
+    Serial.println("[Polaris] Initializing Sensor Board");
+    if (sensorBoard.setup())
+    {
+        Serial.println("[Polaris] Sensor Setup Complete!");
+    }
+    else
+    {
+        Serial.println("[Polaris] Sensor Setup Failed!");
+    }
+    pinMode(6, OUTPUT);
```

Comment:
What is this for?

Reply:
May need to be removed

Reply:
This is the LED pin, we are using it as a status indicator for whether or not the SD card initialized successfully

Reply:
Wheeeeeyyy good job Dashiell!